lib/ddtrace/contrib/mysql2/patcher.rb

```diff
@@ -0,0 +1,73 @@
+module Datadog
+  module Contrib
+    module Mysql2
+      # Patcher enables patching of 'mysql2' client.
+      module Patcher
+        include Base
+        register_as :mysql2
+
+        option :service_name, default: 'mysql2'
+        option :tracer, default: Datadog.tracer
+
+        @patched = false
+
+        def self.patch
+          return @patched if patched?
+
+          require 'ddtrace/ext/app_types'
+          require 'ddtrace/ext/sql'
+
+          patch_mysql2_client()
+
+          @patched = true
+        rescue => e
+          Tracer.log.error("Unable to apply mysql2 integration: #{e}")
+          @patched
+        end
+
+        def self.patched?
+          @patched
+        end
+
+        private_class_method
+
+        def self.patch_mysql2_client
+          ::Mysql2::Client.class_eval do
```

Comment:
Would it be possible to re-implement this as a module instead of with `class_eval`?

The immediate trade off I can think of is that you'd need to use `prepend` which would consequently limit this integration to Ruby 2.0+ (which is reasonable.) The benefit is that this would reduce the chance that another gem conflicts with a method overwrite here, and we'd have a cleaner implementation.